model/src/test/kotlin/licenses/LicenseInfoResolverTest.kt

```diff
@@ -650,6 +736,26 @@ fun containOnlyLicenseSources(vararg licenseSources: LicenseSource): Matcher<Res
         )
     }
 
+fun containLicenseExpressionsExactlyBySource(
+    source: LicenseSource,
+    vararg expressions: SpdxExpression?
+): Matcher<ResolvedLicenseInfo?> =
+    neverNullMatcher { resolvedLicenseInfo ->
+        val actualExpressions = resolvedLicenseInfo.licenses
+            .mapNotNull { it.originalExpressions[source] }
+            .flatten()
+            .toSet()
```

Comment:
Nit: Use `.flattenTo(mutableSetOf())` to avoid a temporary list creation.

Reply:
As we want to use an Immutable Set here. The change would be `.flattenTo(mutableSetOf()).toSet()` which brings no benefit as now a termorary mutable set is created.

Is there a way to get an immutable set without a temporary collection?

Reply:
> Is there a way to get an immutable set without a temporary collection?

Not that I'm aware of.

> Edit: `flattenTo` comes from `org.jetbrains.kotlin.utils.addToStdlib.flattenTo` which sound like a preview feature?!

I don't think so. Preview features are usually hidden behind compiler flags / annotations.

Anyway, it was only a nit and I'm fine with you keeping it as-is, as the performance gain should be negligible for the expected size of the collections. Also, it's only test code.